### ISSUE TITLE
OpenAPI Request Body objects do not follow the spec

### ISSUE THREAD

**Issue by Lucidiot** [encode/django-rest-framework#6562]
On pull request #6532 from @carltongibson:

The OpenAPI 3 specification for a [Request Body Object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#requestBodyObject) says it should follow this format:

``` yaml
requestBody:
  content:
    application/json:
      schema:
        properties:
          something:
            type: string
        required:
        - something
```

However, the generated OpenAPI schemas instead have the following request body objects:

``` yaml
requestBody:
  content:
    application/json:
      properties:
        something:
          type: string
      required:
      - something
```

Probably an easy fix, I may make a PR on that tomorrow.

I also encounter an issue when using [apistar](https://github.com/encode/apistar/) to perform requests with a request body: even when manually fixing the issue in the schema, I still cannot use the request body anywhere because it lacks a custom field that apistar uses [here](https://github.com/encode/apistar/blob/ceefb7b95322b7c2dfb81597438f7800b5eec7b0/apistar/schemas/openapi.py#L459): `x-name`. I guess apistar is to put to blame here, any thoughts? The request bodies apistar expects are:

``` yaml
requestBody:
  x-name: body_name
  content: ......
```

Then requests can be made using `client.request('Endpoint', body_name={...})`

**Comment by carltongibson** [comment]
Hey @Lucidiot. Yes. Looks like I just missed a key. (Should be simple as you say...) 

(I'm back on it myself tomorrow too)

**Comment by carltongibson** [comment]
Closed by #6565
